src/CronExpression.php

```diff
@@ -10,11 +10,13 @@ class CronExpression
     public DaysOfMonthField $day;
     public MonthsField $month;
     public DaysOfWeekField $weekday;
-    public string $locale;
-    public bool $timeFormat24hours;
     public array $translations;
 
-    public function __construct(string $cron, string $locale = 'en', bool $timeFormat24hours = false)
+    /**
+     * @throws CronParsingException
+     * @throws TranslationFileMissingException
+     */
+    public function __construct(string $cron, public string $locale = 'en', public bool $timeFormat24hours = false)
```

Comment:
Could you please revert this change?

I like defining properties in the constructor signature but, for me, it should be all or nothing. I find it confusing when some of the properties are defined in the constructor and some in the class scope.

Reply:
I agree that this is confusing.
I changed it so all paramters use the promoted property constructor. Is that ok for you, or should i still revert it?

Reply:
I'd still rather it's reverted if you don't mind since some properties still have to be declared in the class.

Reply:
Yes of course, i reverted it.